### PR TITLE
Add command to reencrypt specific table/column pair

### PR DESCRIPTION
**Pull request by convenient**
Show that everything is up to date
```bash
$ bin/magento gene:encryption-key-manager:reencrypt-column customer_entity entity_id rp_token
Run with --force to make these changes, this will run in dry-run mode by default
The latest encryption key is number 1, looking for old entries
Looking for 'rp_token' in 'customer_entity', identified by 'entity_id'
No old entries found
```

Then generated a new key
```bash
$ php bin/magento gene:enc:generate --force
The system currently has 2 keys
Generating a new encryption key using the magento core class
_reEncryptSystemConfigurationValues - start
_reEncryptSystemConfigurationValues - end
_reEncryptCreditCardNumbers - start
_reEncryptCreditCardNumbers - end
Cleaning cache
Done
```

Then ran to see a flagged issue that was corrected

```bash
$ bin/magento gene:encryption-key-manager:reencrypt-column customer_entity entity_id rp_token --force
The latest encryption key is number 3, looking for old entries
Looking for 'rp_token' in 'customer_entity', identified by 'entity_id'
########################################################################################################################
entity_id: 967427
ciphertext_old: 2:3:redacted
plaintext: abc123
ciphertext_new: 3:3:redacted
########################################################################################################################
Done
```

Then ran again to verify it was fixed

```bash
$ bin/magento gene:encryption-key-manager:reencrypt-column customer_entity entity_id rp_token
Run with --force to make these changes, this will run in dry-run mode by default
The latest encryption key is number 3, looking for old entries
Looking for 'rp_token' in 'customer_entity', identified by 'entity_id'
No old entries found
```

### CODE REVIEW

File: Console/ReencryptColumn.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            $result = $connection->fetchAll($select);
+            if (empty($result)) {
+                $output->writeln('No old entries found');
+                return Cli::RETURN_SUCCESS;
+            }
+            $connection->beginTransaction();
+            foreach ($result as $row) {
+                $output->writeln(str_pad('', 120, '#'));
+                $output->writeln("$identifier: {$row[$identifier]}");
+                $value = $row[$column];
```
Comment (AlexRapatij) [inline Console/ReencryptColumn.php:131]: Make sense to add something like that for NULL values?
```suggestion
                $value = $row[$column];
                if ($value === null) {
                    $output->writeln("The row #{$row[$identifier]} is NULL, skipped");
                    continue;
                }
```

File: Console/ReencryptColumn.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            /**
+             * @see \Magento\Framework\Model\ResourceModel\Db\AbstractDb::_getLoadSelect()
+             */
+            $tableName = $this->resourceConnection->getTableName($table);
+            $connection = $this->resourceConnection->getConnection();
+            $field = $connection->quoteIdentifier(sprintf('%s.%s', $tableName, $column));
+
+            $select = $connection->select()
+                ->from($tableName, [$identifier, "$column"])
+                ->where("($field LIKE '_:_:____%' OR $field LIKE '__:_:____%')")
```
Comment (AlexRapatij) [inline Console/ReencryptColumn.php:119]: Or to add additional condition here
```suggestion
                ->where("$field IS NOT NULL")
                ->where("($field LIKE '_:_:____%' OR $field LIKE '__:_:____%')")
```

**Review by AlexRapatij — APPROVED** [review]
Ignore my suggestion, NULL-values will be filtered by already existed conditions

File: Console/ReencryptColumn.php (hunk 2 above)
Comment (convenient) [reply Console/ReencryptColumn.php:119]: Also the filtering for nully values on non-indexed tables is longer than selecting the data and filtering in PHP land 👍

**Pull request merged**